js/src/controls/SearchControl.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+const L = require('../leaflet.js');
+const control = require('./Control.js');
+
+export class LeafletSearchControlModel extends control.LeafletControlModel {
+  defaults(){
+    return{
+      _view_name: 'LeafletSearchControlView',
+      _model_name: 'LeafletSearchControlModel',
+      url: null,
+      jsonpParam: 'json_callback',
+      propertyName: 'display_name',
+      propertyLoc: ['lat','lon'],
+      autoType: false,
+      autoCollapse: false,
+      zoom:10,
+      animateLocation:false,
+      marker: L.circleMarker([0,0],{radius:30}),
+    };
+  }
+}
+
+
+export class LeafletSearchControlView extends control.LeafletControlView{
+  initialize(parameters) {
+    super.initialize(parameters);
+    this.map_view = this.options.map_view;
+  }
+
+  create_obj() {
+    this.obj = L.control.search(this.get_options());
+  }
+
+  get_options() {
```

Comment:
You should not need this `get_options`. You can safely remove it once you turned all the attributes in `snake_case`. The `get_options` method is already implemented here: https://github.com/jupyter-widgets/ipyleaflet/blob/3b767b0f4895127c707bee1ee43d089f6f55529b/js/src/utils.js#L17

Reply:
I have removed this now.